core/src/main/java/org/mule/runtime/core/exception/ErrorTypeLocatorFactory.java

```diff
@@ -51,7 +51,7 @@ public static ErrorTypeLocator createDefaultErrorTypeLocator(ErrorTypeRepository
             .addExceptionMapping(RetryPolicyExhaustedException.class, errorTypeRepository.lookupErrorType(RETRY_EXHAUSTED).get())
             .addExceptionMapping(IOException.class, errorTypeRepository.lookupErrorType(CONNECTIVITY).get())
             .addExceptionMapping(SecurityException.class, errorTypeRepository.lookupErrorType(SECURITY).get())
-            .addExceptionMapping(RejectedExecutionException.class, errorTypeRepository.getErrorType(OVERLOAD).get())
+            .addExceptionMapping(SchedulerBusyException.class, errorTypeRepository.getErrorType(OVERLOAD).get())
```

Comment:
the only way we can realistically determine an OVERLOAD is when the runtime executors are full, that's why i changed the thrown exception to this, so we can accurately determine those cases.

In the case of unmanaged executors, it would be responsability of the users of those to handle the RejectedExecutionException.